src/reactA11yAnchorsRule.ts

```diff
@@ -58,7 +58,8 @@ class ReactA11yAnchorsRuleWalker extends ErrorTolerantWalker {
         while (this.anchorInfoList.length > 0) {
             const current: AnchorInfo = this.anchorInfoList.shift();
             this.anchorInfoList.forEach((anchorInfo: AnchorInfo): void => {
-                if (current.href === anchorInfo.href &&
+                if (current.href &&
```

Comment:
Just want to mention that, IMO, if `href` is not defined, the `same href has same text` should not be applied